ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ParticipantPruningIT.scala

```diff
@@ -531,6 +531,58 @@ class ParticipantPruningIT extends LedgerTestSuite {
     }
   })
 
+  test(
+    "PRDivulgenceArchivalPruning",
+    "Prune succeeds for divulgence events whose contracts are archived",
+    allocate(SingleParty, SingleParty),
+    runConcurrently = false, // pruning call may interact with other tests
+  )(implicit ec => { case Participants(Participant(alpha, alice), Participant(beta, bob)) =>
+    for {
+      divulgence <- createDivulgence(alice, bob, alpha, beta)
+      contract <- alpha.create(alice, Contract(alice))
+
+      // Retroactively divulge Alice's contract to bob
+      _ <- alpha.exercise(
+        alice,
+        divulgence.exerciseDivulge(_, contract),
+      )
+
+      _ <- beta.exerciseAndGetContract[Dummy](
+        bob,
+        divulgence.exerciseCanFetch(_, contract),
+      )
+
+      offsetAfterDivulgence <- beta.currentEnd()
+
+      // Dummy create needed to prune after the previous fetch (which also divulges)
+      _ <- beta.create(bob, Dummy(bob))
+
+      _ <- beta.prune(offsetAfterDivulgence, pruneAllDivulgedContracts = false)
+
+      // Bob can still see the divulged contract
+      _ <- beta.exerciseAndGetContract[Dummy](
+        bob,
+        divulgence.exerciseCanFetch(_, contract),
+      )
+
+      offsetAfterDivulgence_2 <- beta.currentEnd()
+
+      _ <- alpha.exercise(alice, contract.exerciseArchive)
+
+      _ <- synchronize(alpha, beta)
+
+      _ <- beta.prune(offsetAfterDivulgence_2, pruneAllDivulgedContracts = false)
+
+      // Bob cannot see the divulged contract
```

Comment:
Should we not check the way in which this fails? 
Since we have not pruned all the divulged contracts, this call will succeed synchronously but fail on the completion stream. This would be different if we pruned the divulged contracts - there the error would be reported synchronously.

Reply:
The purpose of this test is to show that divulgence events are pruned if their associated contract is archived. So the failure happens synchronously. 

I see your point about making sure that the failure is happening during command submission and not due to missing the tree events on the output stream. But I don't see how can this submission pass and the events stream to succeed. In case I understood wrongly, can you clarify?

Reply:
On the other hand, thanks for drawing my attention again to this test. In the original form, it could not pass for privacy-aware ledgers. Changed it to a single-participant test, so it can work on privacy-aware ledgers as well.